Userland/Services/LookupServer/MulticastDNS.h

```diff
@@ -37,7 +37,7 @@ class MulticastDNS : public Core::UDPServer {
     Name m_hostname;
 
     static constexpr sockaddr_in mdns_addr {
-#ifdef AK_OS_BSD_GENERIC
+#if defined(AK_OS_BSD_GENERIC) || defined(AK_OS_GNU_HURD)
```

Comment:
This change seems to confuse clang-format-16. @rymiel Does this one look familiar?

```
andrew@Andrews-MacBook-Pro:~/Source/serenity$ clang-format test.cpp
class C {
    static constexpr sockaddr_in mdns_addr
    {
#if defined(AK_OS_BSD_GENERIC) || defined(AK_OS_GNU_HURD)
        .sin_len = sizeof(struct sockaddr_in),
#endif
        .sin_family = AF_INET,
        // htons(5353)
            .sin_port = 0xe914,
        // 224.0.0.251
            .sin_addr = { 0xfb0000e0 },
        .sin_zero = { 0 }
    }
};
andrew@Andrews-MacBook-Pro:~/Source/serenity$ clang-format --version
clang-format version 16.0.6
```

Reply:
The example you've listed here produces weird indentation, even if you remove the preprocessor directives, but only because you are missing a semicolon at the end of the static variable declaration.
It's still broken if you add the semicolon, but then the preprocessor is required to trigger the bug. And it seems that `#ifdef` doesn't trigger it, which makes it even weirder.
I think this is somewhat related to llvm/llvm-project#56685 but I'll try to have a look at why the two preprocessor directives are handled differently

Reply:
(A fix for this is now at https://github.com/llvm/llvm-project/pull/65409, but you won't see it until clang 18)

Reply:
Great, thanks! I think for this serenity patch the right thing to do is to wrap the member in a `//clang-format off` and reference the upstream issue/PR/expected fix version with a FIXME.

Reply:
Added `//clang-format` directives and issue/PR links, let's see how it fares on the CI